File: python/batch_qa/download_batch_qa_excel.py
import os
from urllib.parse import urljoin

import requests
from utils import MaiAgentHelper

BASE_URL = 'https://api.maiagent.ai/api/v1/'
API_KEY = '<your-api-key>'
WEB_CHAT_ID = '<your-web-chat-id>'
BATCH_QA_FILE_ID = '<your-batch-qa-file-id>'


assert API_KEY != '<your-api-key>', 'Please set your API key'
assert WEB_CHAT_ID != '<your-web-chat-id>', 'Please set your web-chat id'
assert BATCH_QA_FILE_ID != '<your-batch-qa-file-id>', 'Please set your batch qa file id'


def main():
    maiagent_helper = MaiAgentHelper(API_KEY)
    downloaded_file = maiagent_helper.download_batch_qa_excel(WEB_CHAT_ID, BATCH_QA_FILE_ID)
    if downloaded_file:
        print(f'File saved as: {os.path.abspath(downloaded_file)}')


if __name__ == '__main__':
    main()

File: csharp/batch_qa/download_batch_qa_excel.cs

using os;

using Utils;

using System.Diagnostics;

public static class download_batch_qa_excel {
    
    public static string API_KEY = "<your-api-key>";
    
    public static string WEB_CHAT_ID = "<your-web-chat-id>";
    
    public static string BATCH_QA_FILE_ID = "<your-batch-qa-file-id>";
    
    public static void main() {
        var maiagent_helper = MaiAgentHelper(API_KEY);
        var downloaded_file = maiagent_helper.download_batch_qa_excel(WEB_CHAT_ID, BATCH_QA_FILE_ID);
        if (downloaded_file) {
            Console.WriteLine($"File saved as: {os.path.abspath(downloaded_file)}");
        }
    }
    
    static download_batch_qa_excel() {
        main();
    }
    
    static download_batch_qa_excel() {
        Debug.Assert(API_KEY != "<your-api-key>");
        Debug.Assert("Please set your API key");
        Debug.Assert(WEB_CHAT_ID != "<your-web-chat-id>");
        Debug.Assert("Please set your web-chat id");
        Debug.Assert(BATCH_QA_FILE_ID != "<your-batch-qa-file-id>");
        Debug.Assert("Please set your batch qa file id");
        if (@__name__ == "__main__") {
        }
    }
}
